lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -1004,105 +1011,103 @@ export class UserAgentApplication {
+          authenticationRequest.authorityInstance = authority ? AuthorityFactory.CreateInstance(authority, this.validateAuthority) : this.authorityInstance;
+      }
+      // cache miss
+      return authenticationRequest.authorityInstance.ResolveEndpointsAsync()
+      .then(() => {
+        // refresh attept with iframe
+        // Already renewing for this scope, callback when we get the token.
+        if (window.activeRenewals[scope]) {
+          this._logger.verbose("Renew token for scope: " + scope + " is in progress. Registering callback");
+          // Active renewals contains the state for each renewal.
+          this.registerCallback(window.activeRenewals[scope], scope, resolve, reject);
         }
-        // cache miss
-        return authenticationRequest.authorityInstance.ResolveEndpointsAsync()
-        .then(() => {
-          // refresh attept with iframe
-          // Already renewing for this scope, callback when we get the token.
-          if (window.activeRenewals[scope]) {
-            this._logger.verbose("Renew token for scope: " + scope + " is in progress. Registering callback");
-            // Active renewals contains the state for each renewal.
-            this.registerCallback(window.activeRenewals[scope], scope, resolve, reject);
-          }
-          else {
-            if (scopes && scopes.indexOf(this.clientId) > -1 && scopes.length === 1) {
-              // App uses idToken to send to api endpoints
-              // Default scope is tracked as clientId to store this token
-              this._logger.verbose("renewing idToken");
-              this.renewIdToken(scopes, resolve, reject, userObject, authenticationRequest, extraQueryParameters);
-            } else {
-              // renew access token
-              this._logger.verbose("renewing accesstoken");
-              this.renewToken(scopes, resolve, reject, userObject, authenticationRequest, extraQueryParameters);
-            }
+        else {
+          if (scopes && scopes.indexOf(this.clientId) > -1 && scopes.length === 1) {
+            // App uses idToken to send to api endpoints
+            // Default scope is tracked as clientId to store this token
+            this._logger.verbose("renewing idToken");
+            this.renewIdToken(scopes, resolve, reject, userObject, authenticationRequest, extraQueryParameters);
+          } else {
+            // renew access token
+            this._logger.verbose("renewing accesstoken");
+            this.renewToken(scopes, resolve, reject, userObject, authenticationRequest, extraQueryParameters);
           }
-        }).catch((err) => {
-          this._logger.warning("could not resolve endpoints");
-          reject(err);
-          return null;
-        });
-      }
+        }
+      }).catch((err) => {
+        this._logger.warning("could not resolve endpoints");
+        reject(err);
+        return null;
```

Comment:
I don't think returning in here does what you think it does, because this happens asynchronously

Reply:
Not my code.

Reply:
Resolving offline